File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.java
class Solution {
    public int[] applyOperations(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];

        for (int i = 0; i < n - 1; i++) {
            if (nums[i] == nums[i + 1]) {
                nums[i] *= 2;
                nums[i + 1] = 0;
            }
        }
        
        int idx = 0;

        for (int num : nums) {
            if (num != 0) {
                ans[idx++] = num;
            }
        }

        while(idx < n) {
            ans[idx++] = 0;
        }

        return ans;
    }
}

File: 2551-apply-operations-to-an-array/apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = []

        for i in range(n - 1):
            if (nums[i] == nums[i + 1]):
                nums[i] = nums[i] * 2
                nums[i + 1] = 0
        
        for i in nums:
            if i != 0:
                ans.append(i)
        
        while len(ans) < n:
            ans.append(0)

        return ans